core/src/main/scala/org/apache/spark/io/CompressionCodec.scala

```diff
@@ -88,8 +88,12 @@ private[spark] object CompressionCodec {
     } catch {
       case _: ClassNotFoundException | _: IllegalArgumentException => None
     }
-    codec.getOrElse(throw new IllegalArgumentException(s"Codec [$codecName] is not available. " +
-      s"Consider setting $configKey=$FALLBACK_COMPRESSION_CODEC"))
+    codec.getOrElse(throw new SparkIllegalArgumentException(
+      errorClass = "CODEC_NOT_AVAILABLE",
+      messageParameters = Map(
+        "codecName" -> codecName,
+        "configKey" -> configKey,
```

Comment:
Please, wrap it by `toSQLConf`

Reply:
Ditto. Should we move these to spark-core? Or have them duplicated?

Reply:
Oh, sorry. Let's define in `core` at leats `quoteByDefault()`, `toConf`, and `toConfVal`.

Reply:
Done.